yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,396 @@
+    ///         last time its core position(s) were adjusted. Examples include
+    ///         unwrapping extra rewards. This call is only used during normal
+    ///         operation of a strategy, and should be optimized to minimize
+    ///         losses as much as possible. This strategy implements the
+    ///         aforementioned behavior by getting KEEP tokens from the Saddle
+    ///         reward pool, using obtained tokens to buy one of the Saddle
+    ///         pool's accepted token and depositing that token back to the
+    ///         Saddle pool. This way the strategy is gaining new vault's
+    ///         underlying tokens thus making the profit.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    /// @return profit Amount of realized profit.
+    /// @return loss Amount of realized loss.
+    /// @return debtPayment Amount of repaid debt. The value of debtPayment
+    ///         should be less than or equal to debtOutstanding. It is okay for
+    ///         it to be less than debtOutstanding, as that should only used as
+    ///         a guide for how much is left to pay back. Payments should be
+    ///         made to minimize loss from slippage, debt, withdrawal fees, etc.
+    function prepareReturn(uint256 debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 profit,
+            uint256 loss,
+            uint256 debtPayment
+        )
+    {
+        // Get the initial balance of the vault's underlying token under
+        // strategy management.
+        uint256 initialWantBalance = balanceOfWant();
+
+        // Get KEEP rewards from the Saddle reward pool.
+        ILPRewards(tbtcSaddleLPRewards).getReward();
+
+        // Buy WBTC using obtained KEEP tokens.
+        uint256 keepBalance = IERC20(keep).balanceOf(address(this));
+        if (keepBalance > 0) {
+            IERC20(keep).safeIncreaseAllowance(uniswap, keepBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = keep;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(uniswap).swapExactTokensForTokens(
+                keepBalance,
+                0,
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Deposit acquired WBTC to the Saddle pool to gain additional
+        // vault's underlying tokens.
+        uint256 wbtcBalance = IERC20(wbtc).balanceOf(address(this));
+        if (wbtcBalance > 0) {
+            IERC20(wbtc).safeIncreaseAllowance(tbtcSaddlePoolSwap, wbtcBalance);
+
+            uint256[] memory amounts = new uint256[](4);
+            amounts[1] = wbtcBalance;
+
+            bytes32[] memory merkleProof = new bytes32[](1);
```

Comment:
Don't we need any value here?

Reply:
According to [Saddle-finances contract](https://github.com/saddle-finance/saddle-contract/blob/2b6c3c2a25b1b78d62cb048cbbc6bf0700e2f711/contracts/guarded/SwapGuarded.sol#L429) this parameter is ignored when the pool is not guarded:
"""
@param merkleProof data generated when constructing the allowlist merkle tree. Users can
get this data off chain. Even if the address is in the allowlist, users must include
a valid proof for this call to succeed. If the pool is no longer in the guarded release phase,
this parameter is ignored.
"""

And their [contract](https://etherscan.io/address/0x4f6A43Ad7cba042606dECaCA730d4CE0A57ac62e) is indeed not guarded, so I think we can ignore this parameter, I simply used:
'''"
new bytes32[](0) // ignored
"""